hello/hello_continue_as_new.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+from dataclasses import dataclass
+from datetime import timedelta
+
+from temporalio import activity, workflow
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+
+@dataclass
+class ComposeGreetingInput:
+    greeting: str
+    name: str
+
+
+@activity.defn
+async def compose_greeting(input: ComposeGreetingInput) -> str:
+    return f"{input.greeting}, {input.name}!"
+
+
+@workflow.defn
+class GreetingWorkflow:
+    def __init__(self) -> None:
+        self.exit_requested = False
+
+    @workflow.run
+    async def run(self, name: str) -> str:
+        # Invoke activity (or do other meaningful work)
+        result = await workflow.execute_activity(
+            compose_greeting,
+            ComposeGreetingInput("Hello", name),
+            start_to_close_timeout=timedelta(seconds=10),
+        )
+
+        # Continue as new every 2 seconds until we are told to complete. This is
+        # just a demonstration. In production code you might want to continue as
+        # new when the history is growing too large. Continue as new is
+        # essentially a workflow restart (though another workflow can be used).
+        try:
+            await workflow.wait_condition(lambda: self.exit_requested, timeout=2)
```

Comment:
Hmm.. timeout should be a timedelta or the are name should have a unit.

Reply:
This is built to match https://docs.python.org/3/library/asyncio-task.html#asyncio.wait. Really, none of the Python APIs use `timedelta`s (https://docs.python.org/3/library/asyncio-task.html#asyncio.sleep, https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.call_later, etc).

I am trying to avoid accepting `Union[float, timedelta]` everywhere to avoid confusion, so I figured following other `asyncio` APIs on just this call makes sense. But I can change if wanted. (I wish we had reviewed these before)

Reply:
I wish we had reviewed this earlier too, seeing these samples now puts this in my face so more suggestions come up.

Nothing critical, if matching asyncio APIs is a goal then I'd say it's fine as is.